mobile/lib/src/infrastructure/connection_repositories/bluetooth_repository.dart

```diff
@@ -19,43 +20,48 @@ class BluetoothRepository implements IAcquisitionDeviceRepository {
   FlutterReactiveBle flutterReactiveBle;
   StreamSubscription<ConnectionStateUpdate> _connectedDeviceStream;
   StreamSubscription<DiscoveredDevice> _bluetoothScanSubscription;
-  final List<AcquisitionDevice> _acquisitionDevicePersistency = [];
-  final streamController = StreamController<List<AcquisitionDevice>>();
+  final streamController = StreamController<AcquisitionDevice>();
 
   @override
-  void initializeRepository() {
+  Stream<AcquisitionDevice> scan() {
     if (_bluetoothScanSubscription == null) {
-      flutterReactiveBle = FlutterReactiveBle();
-
-      _bluetoothScanSubscription = flutterReactiveBle.scanForDevices(
-          withServices: []).listen((device) => addDevice(device));
+      _initScan();
     } else {
-      _bluetoothScanSubscription.resume();
-      _acquisitionDevicePersistency.clear();
+      resumeScan();
     }
+
+    return streamController.stream;
   }
 
   void addDevice(DiscoveredDevice bluetoothDevice) {
     var device = AcquisitionDevice(
-        UniqueId.from(bluetoothDevice.id.toString()), bluetoothDevice.name);
+        UniqueId.from(bluetoothDevice.id),
+        (bluetoothDevice.name.isEmpty) ? 'Unknown' : bluetoothDevice.name,
+        DeviceType.bluetooth);
 
-    final idx = _acquisitionDevicePersistency.indexOf(device);
+    streamController.add(device);
+  }
 
-    if (idx == -1) {
-      _acquisitionDevicePersistency.add(device);
-    } else {
-      _acquisitionDevicePersistency[idx] = device;
-    }
+  void _initScan() {
+    flutterReactiveBle = FlutterReactiveBle();
 
-    streamController.add(_acquisitionDevicePersistency);
+    _bluetoothScanSubscription = flutterReactiveBle
+        .scanForDevices(withServices: []).listen((device) => addDevice(device));
```

Comment:
à la place d'utiliser listen et addDevice, on pourrait utiliser plutôt un map. On aurait plus besoin d'utiliser un stream controller et le code serait encore plus simple.

```suggestion
        .scanForDevices(withServices: []).map((device) => AcquisitionDevice(
            UniqueId.from(bluetoothDevice.id),
            (bluetoothDevice.name.isEmpty) ? 'Unknown' : bluetoothDevice.name,
            DeviceType.bluetooth,
        );
```